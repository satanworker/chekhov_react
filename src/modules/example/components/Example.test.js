import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import Example from './Example'

const defaultProps = {
  count: 0,
  text: '',
  random: 0,
  onClick: () => {},
  onChange: () => {},
  t: () => ''
}

const Component = () => (
  <MemoryRouter>
    <Example { ...defaultProps } />
  </MemoryRouter>
)

it('renders Example without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Component />, div)
})

it('renders correctly', () => {
  const tree = renderer.create(<Component />).toJSON()
  expect(tree).toMatchSnapshot()
})
