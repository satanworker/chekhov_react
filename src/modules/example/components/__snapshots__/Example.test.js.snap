// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa knYxsw"
>
  <h2>
    
    : 
    <b
      id="click-counter"
    >
      0
    </b>
  </h2>
  <button
    className="sc-bwzfXH ePvrkd"
    id="counter"
    onClick={[Function]}
  >
    Dont click!
  </button>
  <hr />
  <h2>
    Text: 
    <b>
      
    </b>
  </h2>
  <input
    className="sc-htpNat cEZYLy"
    onChange={[Function]}
    value=""
  />
  <hr />
  <h2>
    Saga random value: 
    <b>
      0.0000
    </b>
  </h2>
  <hr />
  <a
    href="/"
    onClick={[Function]}
  >
    Main
  </a>
</div>
`;
